cpp/proto/substrait/extension_rels.proto

```diff
@@ -58,3 +58,16 @@ message NamedTapRel {
   // If empty, field names will be automatically generated.
   repeated string columns = 3;
 }
+
+message SegmentedAggregateRel {
+  substrait.RelCommon common = 1;
+
+  // Grouping keys of the aggregation
+  repeated substrait.Expression.ReferenceSegment grouping_keys = 2;
+
+  // Segment keys of the aggregation
+  repeated substrait.Expression.ReferenceSegment segment_keys = 3;
+
+  // A list of one or more aggregate expressions along with an optional filter.
+  repeated substrait.AggregateRel.Measure measures = 4;
```

Comment:
Not sure about reusing AggregationRel.Measure here, it feels to me this introduced an dependency between AggregationRel and SegmentedAggregateRel which we wanted to avoid? Maybe we should have SegmentedAggregateRel.Measure here - @westonpace WDYT?

Reply:
Note that this [was suggested](https://github.com/apache/arrow/pull/34627#discussion_r1145005055) by @westonpace.

Reply:
I think this is ok.  If this were in the Substrait spec we might separate the two so we could allow the two implementations to diverge in the future.  However, for something internal I think it is fine to keep it simpler.

Reply:
kk sounds good to me.